shell/plugins/internal-api/shell/shell.api.ts

```diff
@@ -0,0 +1,108 @@
+import { GrowlConfig } from '@shell/types/internal-api/shell/growl';
+import { ModalConfig } from '@shell/types/internal-api/shell/modal';
+import { SlideInConfig } from '@shell/types/internal-api/shell/slideIn';
+
+import { BaseApi } from '@shell/plugins/internal-api/shared/base-api';
+
+export default class ShellApi extends BaseApi {
+  static apiName() {
+    return 'shell';
+  }
+
+  /**
+   * Dispatches a growl notification.
+   *
+   * @param config - Configuration for the growl notification.
+   * - If `message` is a string, it is treated as the main content of the notification.
+   * - If `message` is a `DetailedMessage` object, `title` and `description` are extracted.
+   *
+   * Example:
+   * ```ts
+   * this.$shell.growl({ message: 'Operation successful!', type: 'success' });
+   * this.$shell.growl({ message: { title: 'Warning', description: 'Check your input.' }, type: 'warning' });
+   * ```
+   */
+  protected growl(config: GrowlConfig): void {
```

Comment:
I think we may want these to be verbs instead of nouns since an action should occur when these are invoked but it's a pretty minor detail so I can go either way.

Reply:
That's a good point. In the future we can update these to use verbs, but I think for the present this should suffice.